Helper/Cart.php

```diff
@@ -1357,14 +1453,15 @@ public function getCartData($paymentOnly, $placeOrderPayload, $immutableQuote =
         // Process Mirasvit Store Credit
         /////////////////////////////////////////////////////////////////////////////////
         if ($this->discountHelper->isMirasvitStoreCreditAllowed($immutableQuote)){
-            $amount = $this->discountHelper->getMirasvitStoreCreditAmount($immutableQuote, $paymentOnly);
+            $amount = abs($this->discountHelper->getMirasvitStoreCreditAmount($immutableQuote, $paymentOnly));
             $roundedAmount = $this->getRoundAmount($amount);
             $cart['discounts'][] = [
                 'description' => 'Store Credit',
                 'amount'      => $roundedAmount,
                 'type'        => 'fixed_amount',
             ];
 
+            $diff -= $amount * 100 - $roundedAmount;
```

Comment:
Shouldn't this be `$diff = $amount * 100 - $roundedAmount;`? Same for the other diffs down below.

Reply:
@matt-thomason for the items, shipping and tax prices this $diff collector increment is positive, for the discounts is negative. The same is with the $totalAmount aggregator.

Reply:
Gotcha 👀